### PR TITLE
Docs with sphinx for HC3

### LINKED ISSUE #54: Include terraform into the index of docsportal
Here are 2 points to be considered independent from each other:
- evaluate whether we can/it makes sense at all to publish build TF documentation directly to HC. Currently we only have link to TF registry (but maybe this is even ok)
- include TF into the search of HC so that we also promote TF in search results

### LINKED ISSUE #121: Include TF docs in HC

### PR DESCRIPTION
**Pull request by SebastianGode**
## Summary of the Pull Request

Adds required files and entries to be able to produce doc output for our HC3.

image

### CODE REVIEW

**Comment by SebastianGode** [comment]
It could break your public documentation a bit as the toctree inside the index.md will likely be rendered there as well and that will look ugly.
Don't know if there's a good other solution.
https://github.com/opentelekomcloud/terraform-provider-opentelekomcloud/blob/docs-sphinx/docs/index.md

**Comment by anton-sidelnikov** [comment]
Hello @SebastianGode, is it possible to keep current structure like in https://registry.terraform.io/providers/opentelekomcloud/opentelekomcloud/latest/docs ?

Current docs reads subcategory field and put docs in same place if category match, and the divide them on resources and data-sources:
image

**Comment by anton-sidelnikov** [comment]
maybe toctree will not renders if you put it in between ---
--- {toctree} ---

File: docs/data-sources/index.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Data Sources
```
Comment (anton-sidelnikov) [inline docs/data-sources/index.md]: is it possible to build this automaticaly?

File: docs/resources/index.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
+# Resources
+
```
Comment (anton-sidelnikov) [inline docs/resources/index.md]: and this?

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  sh
+commands =
+  sphinx-build -W --keep-going -b json docs/ docs/build/json/
+  # Drop data useless for the search - wrap it also with sh/xargs due to bugs
+  # in tox
+  sh -c "find docs/build/json -type d -and '(' -name '_images' -or -name '_static' -or -name '_sources' ')' -print0 | xargs -0 rm -rf"
+
+[doc8]
+ignore = D001
+extensions = .rst, .yaml
```
Comment (anton-sidelnikov) [inline tox.ini]: end line

File: .gitignore [hunk 4]
```diff
@@ -28,3 +28,7 @@ localrc
 
 dist
 releasenotes/build
+
+docs/build
+.tox/
+terraform_provider_opentelekomcloud.egg-info/
```
Comment (anton-sidelnikov) [inline .gitignore]: end line

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: docs/data-sources/index.md (hunk 1 above)
Comment (SebastianGode) [reply docs/data-sources/index.md]: I have added an autogeneration in the next commit.

File: docs/resources/index.md (hunk 2 above)
Comment (SebastianGode) [reply docs/resources/index.md]: I have added an autogeneration in the next commit.

**Comment by SebastianGode** [comment]
> maybe toctree will not renders if you put it in between --- --- {toctree} ---

This sadly doesn't work.
We could add it on build time, but I'm not sure on how to do this without making git aware of it.

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by artem-lifshits — DISMISSED** [review]

**Review by vladimirvshivkov — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Review by artem-lifshits — APPROVED** [review]

**Pull request merged**